### PR TITLE
IOS/ES: Implement ES_DeleteSharedContent

### PR DESCRIPTION
**Pull request by leoetlino**
Used by the Wii Shop in some cases. Implementing this also gets us closer to 100% completion.

### CODE REVIEW

File: Source/Core/Core/IOS/ES/TitleManagement.cpp [hunk 1]
```diff
@@ -622,6 +622,51 @@ IPCCommandResult ES::ExportTitleDone(Context& context, const IOCtlVRequest& requ
...
+    return ES_EIO;
+
+  return IPC_SUCCESS;
+}
+
+IPCCommandResult ES::DeleteSharedContent(const IOCtlVRequest& request)
+{
+  if (!request.HasNumberOfValidVectors(1, 0) || request.in_vectors[0].size != 20)
+    return GetDefaultReply(ES_EINVAL);
+  std::array<u8, 20> sha1;
```
Comment (BhaaLseN) [inline Source/Core/Core/IOS/ES/TitleManagement.cpp]: Maybe move this up two lines and use `request.in_vectors[0].size != sha1.size()` instead?

File: Source/Core/Core/IOS/ES/TitleManagement.cpp [hunk 2]
```diff
@@ -622,6 +622,51 @@ IPCCommandResult ES::ExportTitleDone(Context& context, const IOCtlVRequest& requ
 {
   return GetDefaultReply(ExportTitleDone(context));
 }
+
+ReturnCode ES::DeleteSharedContent(const std::array<u8, 20>& sha1) const
+{
+  IOS::ES::SharedContentMap map{Common::FromWhichRoot::FROM_SESSION_ROOT};
+  const std::string content_path = map.GetFilenameFromSHA1(sha1);
+  if (content_path == "unk")
```
Comment (BhaaLseN) [inline Source/Core/Core/IOS/ES/TitleManagement.cpp]: Maybe a little late to the party (and totally ignorable), but that magic string `"unk"` appears in at least 3 other places. Perhaps its time to make it a constant or something like that?

Comment (leoetlino) [inline Source/Core/Core/IOS/ES/TitleManagement.cpp]: Yeah, I don't like the magic string either. I'm just waiting for std::optional to drop that ;)

File: Source/Core/Core/IOS/ES/TitleManagement.cpp (hunk 1 above)
Comment (leoetlino) [inline Source/Core/Core/IOS/ES/TitleManagement.cpp]: Sure.

File: Source/Core/Core/IOS/ES/TitleManagement.cpp (hunk 2 above)
Comment (BhaaLseN) [inline Source/Core/Core/IOS/ES/TitleManagement.cpp]: Alright, good enough for me.

**Pull request merged**